lib/src/trim_editor.dart

```diff
@@ -270,139 +350,101 @@ class _TrimEditorState extends State<TrimEditor> with TickerProviderStateMixin {
+  /// Called during dragging, only executed if [_allowDrag] was set to true in
+  /// [_onDragStart].
+  /// Makes sure the limits are respected.
+  void _onDragUpdate(DragUpdateDetails details) {
+    if (!_allowDrag) return;
+
+    _circleSize = widget.circleSizeOnDrag;
+
+    if (_dragType == EditorDragType.left) {
+      if ((_startPos.dx + details.delta.dx >= 0) &&
+          (_startPos.dx + details.delta.dx <=
+              _endPos.dx) &&
+          !(_endPos.dx - _startPos.dx - details.delta.dx > maxLengthPixels!)) {
+        _startPos += details.delta;
+        _onStartDragged();
+      }
+    } else if (_dragType == EditorDragType.center) {
+      if ((_startPos.dx + details.delta.dx >= 0) &&
+          (_endPos.dx + details.delta.dx <= _thumbnailViewerW)) {
+        _startPos += details.delta;
+        _endPos += details.delta;
+        _onStartDragged();
+        _onEndDragged();
+      }
+    } else {
+      if ((_endPos.dx + details.delta.dx <= _thumbnailViewerW) &&
+          (_endPos.dx + details.delta.dx >=
+              _startPos.dx) &&
+          !(_endPos.dx - _startPos.dx + details.delta.dx > maxLengthPixels!)) {
+        _endPos += details.delta;
+        _onEndDragged();
       }
     }
+    setState(() {});
+  }
 
-    _initializeVideoController();
-    _endPos = Offset(
-      maxLengthPixels != null ? maxLengthPixels! : _thumbnailViewerW,
-      _thumbnailViewerH,
-    );
-
-    // Defining the tween points
-    _linearTween = Tween(begin: _startPos.dx, end: _endPos.dx);
+  void _onStartDragged() {
+    _startFraction = (_startPos.dx / _thumbnailViewerW);
+    _videoStartPos = _videoDuration * _startFraction;
+    widget.onChangeStart!(_videoStartPos);
+    _linearTween.begin = _startPos.dx;
+    _animationController!.duration =
+        Duration(milliseconds: (_videoEndPos - _videoStartPos).toInt());
+    _animationController!.reset();
+  }
 
-    _animationController = AnimationController(
-      vsync: this,
-      duration: Duration(milliseconds: (_videoEndPos - _videoStartPos).toInt()),
-    );
+  void _onEndDragged() {
+    _endFraction = _endPos.dx / _thumbnailViewerW;
+    _videoEndPos = _videoDuration * _endFraction;
+    widget.onChangeEnd!(_videoEndPos);
+    _linearTween.end = _endPos.dx;
+    _animationController!.duration =
+        Duration(milliseconds: (_videoEndPos - _videoStartPos).toInt());
+    _animationController!.reset();
+  }
 
-    _scrubberAnimation = _linearTween.animate(_animationController!)
-      ..addListener(() {
-        setState(() {});
-      })
-      ..addStatusListener((status) {
-        if (status == AnimationStatus.completed) {
-          _animationController!.stop();
-        }
-      });
+  /// Drag gesture ended, update UI accordingly.
+  void _onDragEnd(DragEndDetails details) {
+    setState(() {
+      _circleSize = widget.circleSize;
+      videoPlayerController
+          .seekTo(Duration(milliseconds: _videoStartPos.toInt()));
+    });
```

Comment:
Here can be one improvement, you are just updating the video to the start position of the slider even if the right end is dragged. So, it would be nice that you use the `enum` to check which end is dragged and set the position accordingly.

Like set the video position to end slider position if the right end is dragged:

```dart
      if (_dragType == EditorDragType.left) {
        videoPlayerController
            .seekTo(Duration(milliseconds: _videoStartPos.toInt()));
      } else if (_dragType == EditorDragType.right) {
        videoPlayerController
            .seekTo(Duration(milliseconds: _videoEndPos.toInt()));
      } else {
        videoPlayerController
            .seekTo(Duration(milliseconds: _videoStartPos.toInt()));
      }
```

Reply:
Fine with me, the behaviour is a bit different, but in a way I don't mind.